load_test.go

```diff
@@ -1073,3 +1074,359 @@ func Test_validatePatch(t *testing.T) {
+						},
+						"also_perfectly_valid": {
+							Paths: []string{"/also_valid"},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "invalid SymlinkTarget should fail validation: all symlink 'newpaths' should be unique(path)",
+			ImageConfig: ImageConfig{
+				Post: &PostInstall{
+					Symlinks: map[string]SymlinkTarget{
+						"perfectly_valid": {
+							Path: "/also_valid",
+						},
+						"also_perfectly_valid": {
+							Path: "/also_valid",
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "invalid SymlinkTarget should fail validation: path and paths are mutually exclusive",
+			ImageConfig: ImageConfig{
+				Post: &PostInstall{
+					Symlinks: map[string]SymlinkTarget{
+						"perfectly_valid": {
+							Path:  "/also_valid",
+							Paths: []string{"/also_valid_too", "also_valid_too,_also"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range table {
+		t.Run(test.desc, func(t *testing.T) {
+			symlinks := test.ImageConfig.Post.Symlinks
+
+			err := validateSymlinks(symlinks)
+			if err == nil {
+				if test.shouldPassVaildation {
+					return
+				}
+
+				t.Error("should not have passed validation, but did anyway")
+				return
+			}
+
+			// err is non-nil
+
+			if test.shouldPassVaildation {
+				t.Errorf("should have passed validation, but failed with error: %s\n%#v\n", err, symlinks)
+				return
+			}
+		})
+	}
+}
+
+func TestImageFillDefaults(t *testing.T) {
+	t.Run("postinstall", testPostInstallFillDefaults)
+}
+
+func testPostInstallFillDefaults(t *testing.T) {
+	t.Run("symlinks", testSymlinkFillDefaults)
+}
+
+func testSymlinkFillDefaults(t *testing.T) {
+	type tableEntry struct {
+		desc   string
+		input  ImageConfig
+		output ImageConfig
+	}
+
+	// note: fillDefaults is run after validation, so input is assumed to be
+	// valid
+
+	table := []tableEntry{
+		{
+			desc: "empty Path and single Paths should remain untouched",
+			input: ImageConfig{
+				Post: &PostInstall{
+					Symlinks: map[string]SymlinkTarget{
+						"oldpath": {
+							Path:  "",
+							Paths: []string{"/newpath"},
+						},
+					},
+				},
+			},
+			output: ImageConfig{
+				Post: &PostInstall{
+					Symlinks: map[string]SymlinkTarget{
+						"oldpath": {
+							Path:  "",
+							Paths: []string{"/newpath"},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "path should be moved to Paths",
+			input: ImageConfig{
+				Post: &PostInstall{
+					Symlinks: map[string]SymlinkTarget{
+						"oldpath": {
+							Path:  "/newpath2",
+							Paths: []string{"/newpath1"},
+						},
+					},
+				},
+			},
+			output: ImageConfig{
+				Post: &PostInstall{
+					Symlinks: map[string]SymlinkTarget{
+						"oldpath": {
+							Path:  "",
+							Paths: []string{"/newpath1", "/newpath2"},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "Path should be appended to Paths",
+			input: ImageConfig{
+				Post: &PostInstall{
+					Symlinks: map[string]SymlinkTarget{
+						"oldpath": {
+							Path:  "/newpath3",
```

Comment:
I don't think we need to test this since its not a valid case.

Reply:
Good catch. Fixed it so the test code will reject invalid tests.